src/control/server/engine/utils.go

```diff
@@ -281,6 +281,7 @@ func getLogLevelAssignments(masks, subsystemsStr string, baseLevel LogLevel) ([]
 		})
 	}
 
+	// Build subsystems slice from input string.
 	var subsystems []string
 	for _, ss := range strings.Split(subsystemsStr, logMasksStrAssignSep) {
 		if ss == "" || strings.ToUpper(ss) == "ALL" {
```

Comment:
When we find ALL, I was expecting to break the loop and reset `subsystems` to return all the assignments thanks to line 293.
For example (from what I understand), with 'vos,all,dmg' subsystems we will not have all the assignments but only the `vos` and `dmg` ones.

Reply:
Its a good question, the implementation is okay though because `server/engine/utils.go:validateLogNames` will make sure that ALL|all is the only token in the string and so `vos,all,dmg` is refused as an illegal subsystems string. `all` will be parsed as "" and is treated by the d_log api as `all` (which is the default for empty/unset value).